### PR TITLE
Read points in get_entity_by_ref_tool; reject non-computable roles upfront (2.3.1)

### PR DESCRIPTION
**Pull request by hemati**
## Summary

Two related changes uncovered by a prod-bug investigation against the `wahed` project, where every `set_userstory_points_tool` call was returning HTTP 500 with the cryptic `Invalid role id '221'` server-side error from Taiga.

## Root cause of the prod bug (the real one)

Probed live `wahed`:

```
roles:
  199 Product Owner   computable=False
  200 Stakeholder     computable=False
  221 Developer       computable=False
```

All three roles in `wahed` have `computable=False`. Taiga's userstory PATCH validator rejects any role-ID whose `role.computable` is False, but it returns the misleading `Invalid role id '<id>'` — not "role not configured for points estimation". The role IS valid; it's just not enabled for story-point estimation. Project admins fix this in Taiga UI → Members → Roles → "Compute story points for this role".

Code-side fix: catch this client-side instead of letting the user chase payload-shape ghosts.

### `set_userstory_points_tool` — computable-validation gate

```python
if not getattr(role, "computable", True):
    non_computable_roles.append(role_name)
    continue
```

400 response on rejection now carries:
- `non_computable_roles: ["Developer"]` — the rejected names
- `computable_roles: []` — what the project actually allows (empty here, signalling no fallback exists)
- The standard `available_roles` and `available_point_values` for unrelated diagnostics
- A remediation message pointing at `Taiga project admin → Members/Roles → "Compute story points for this role"`

Docstring updated to mention the `computable` requirement upfront so the LLM steers users correctly without hitting the 400.

### `get_entity_by_ref_tool` — read the per-role points

User-story responses now include a `points` field:

```json
"points": {"Developer": 5, "UX": 2}
```

Shape is symmetric to `set_userstory_points_tool`'s input — a read + write round-trip stays in the same vocabulary. Logic lives in a new helper `_format_userstory_points(entity, project)` so the ID-to-name resolution is testable without spinning up the full get-by-ref machinery (status lookups, custom-attribute fetches, history pagination).

Edge cases the helper silently drops:
- Stale role-ids (role-id key in `us.points` no longer in `project.list_roles()`)
- Assignments pointing to the `?` (unestimated, `value=None`) point

Consumers see only role-name keys with concrete numeric values — never partial or null entries.

## Why both in one PR

They're complementary: the new read path lets the same caller diagnose what's currently set on a story before deciding what to write, and the computable-validation surfaces the failure mode the user just hit. Bundling avoids two close-spaced 2.3.x releases.

## Validation

- `make test` (or conda equivalent): **171 passed, 1 skipped** (was 167 before this PR).
- New tests:
  - `test_non_computable_role_returns_400_with_diagnostic` — locks the new client-side rejection path.
  - `test_format_userstory_points_returns_role_name_to_value` — happy path.
  - `test_format_userstory_points_empty_when_unset` — `entity.points = None` → `{}`.
  - `test_format_userstory_points_drops_stale_role_id_and_unestimated` — both edge cases in one test.
- Existing tests untouched (the new computable check defaults to `True` via `getattr`, so pre-existing fakes without the attr still pass).

## Notable

- **Version bump 2.3.0 → 2.3.1**: patch, because the changes are additive (new response field, new error path) and don't break any existing caller.
- **Release flow**: `ci_publish.yml` auto-publishes to PyPI on merge. After merge, the `taiga` repo's `Jenkinsfile` doesn't strictly need a re-bump (`TAIGA_MCP_VERSION` is already at `2.3.0`, which satisfies most callers); but if you want the deployed `taiga-mcp` to ship 2.3.1, bump default → `2.3.1` in `Shikenso-Analytics/taiga` and trigger one explicit Jenkins build to seed the param-cache.
- **User action still required for `wahed`**: enable `computable=True` on the Developer role via the Taiga UI before retrying `set_userstory_points_tool` — this PR makes the failure mode loud, not the project setting.

## Test plan

- [x] Targeted: `pytest tests/unit_tests/test_set_userstory_points_tool.py tests/unit_tests/test_get_entity_by_ref_tool.py -v` (25/25 pass).
- [x] Full suite: `env -u TAIGA_URL pytest tests/unit_tests/` (171 pass).
- [ ] Live smoke after enabling computable on Developer in `wahed` UI: round-trip `get_entity_by_ref_tool` → `set_userstory_points_tool({"Developer": 5})` → `get_entity_by_ref_tool` and verify the field is updated.

### CODE REVIEW

File: langchain_taiga/tools/taiga_tools.py [hunk 1]
```diff
@@ -1461,7 +1488,9 @@ def get_entity_by_ref_tool(project_slug: str, entity_ref: int, entity_type: str)
         watchers = [get_user(w) for w in watchers]
     result["watchers"] = watchers
 
-    # For userstories, include the count of related tasks.
+    # For userstories, include the count of related tasks AND the
+    # per-role story points (symmetric to set_userstory_points_tool's
+    # input shape: {"Developer": 5, "UX": 2}).
```
[the remark this answers is not in the document]
Comment (hemati) [reply langchain_taiga/tools/taiga_tools.py:1499]: Fixed in commit `18a299c`. The example response in `get_entity_by_ref_tool`'s docstring now includes the `points` field with a comment that it's userstory-only and references `set_userstory_points_tool`'s symmetric input shape — so an LLM reading the docstring sees the field is intentional.

File: langchain_taiga/tools/taiga_tools.py [hunk 2]
```diff
@@ -2673,6 +2703,13 @@ def set_userstory_points_tool(
     (unestimated) point has value=None and cannot be set via this tool;
     use the Taiga UI for that.
 
+    The target role MUST be configured as **computable** in the project
+    (Taiga admin → Members/Roles → "Compute story points for this role").
+    Non-computable roles are rejected by Taiga's userstory PATCH endpoint
+    with a generic ``Invalid role id`` error; this tool surfaces that
+    upfront as a 400 with ``non_computable_roles`` so the user knows to
+    flip the project setting.
```
[the remark this answers is not in the document]
Comment (hemati) [reply langchain_taiga/tools/taiga_tools.py:2717]: Fixed in commit `18a299c`. The `Returns:` section now explicitly enumerates each diagnostic field the 400 response can carry (`unresolved_roles`, `unresolved_values`, `non_computable_roles`, `available_roles`, `computable_roles`, `available_point_values`) with one-line semantics for each. The intent is for a calling LLM to be able to branch its retry strategy by which list is non-empty (e.g. "non_computable_roles set → tell user to enable computable in Taiga UI" vs "unresolved_roles set → suggest from available_roles") without trial and error.
